web-ui/src/api/notifications.js

```diff
@@ -83,4 +83,17 @@ export const emailSupervisorAssignment = async (member, cookie) => {
   } else {
     console.warn("Unable to send email regarding as member was not valid and missing required fields", member)
   }
+}
+export const emailGuildLeaders = async (members, guild, cookie) => {
+  if (members) {
+    for (const member of members) {
+      if (member.workEmail && guild?.name) {
+        await sendEmail("You have been assigned as a guild leader of " + guild.name,
+            "Congratulations you have been assigned as a guild leader of " + guild.name,
+            false, [member.workEmail], cookie)
+      } else {
+        console.warn("Unable to send guild leader email regarding as member was not valid and missing required fields", member)
```

Comment:
Since we only iterate over `members` when it's not empty, and because we iterate over all members without the need to break out of the loop, a `map` or `forEach` can be used flatten nesting like:

```js
  members.forEach(async member => {
    if (!member.workEmail) {
      console.warn("Unable to send guild leader email as member is missing required fields", member);
      return;
    }

    const subject = `You have been assigned as a guild leader of ${guild.name}`;
    const body = `Congratulations, you have been assigned as a guild leader of ${guild.name}`;
    await sendEmail(subject, body, false, [member.workEmail], cookie);
  });
  ```